### PR TITLE
Remove user id from filepath

### LINKED ISSUE #1097: Modify filepath not storing the user id part (to be continued)
### Note that changes to the first step makes all other things break, so this has to be kept on a separate branch until it can be merged into the master branch.

As a developer
I want to change the way the filepath is stored in database (remove the user id part)
so that the user is not part of the filepath

### Note

Changes required in these places:
- s3inbox should not store the userid as part of the filename in the database
  - the message to cega should not include the userid in the filepath (no message should include the userid in the filepath).
- ingest needs to be able to find the file, so it has to concatenate the userid
- mapper needs to be able to remove the file from the inbox
- download - no change if it's fixed above
  - s3
  - encrypted s3
  - decrypted REST

### PR DESCRIPTION
**Pull request by MalinAhlberg**
**Related issue(s) and PR(s)**  
This PR closes #1097 .

**Description**
The user's id is no longer stored in the database, so that it will never be shown to a user of download.
- filepaths in integration tests are updated to reflect this change
- inbox does not store the user id
- ingest and mapper do not expect the user id to be there

**How to test**
- all tests look ok and pass
- the endpoints in #1095 do not leak user information
- download still works

**Note**
The db schema version from #1123 should be set, but potentially in another PR.
The base of this PR is `feature/do-not-leak-user-ids`, which is branched off main. I do not feel strongly about the naming of this branch, so feel totally free to have opinions.

### CODE REVIEW

**Review by jbygdell — APPROVED** [review]

File: .github/integration/tests/sda/60_api_admin_test.sh [hunk 1]
```diff
@@ -104,27 +101,18 @@ fi
...
 s3cmd -c s3cfg put NA12878.bam.c4gh s3://test_dummy.org/NE12878.bam.c4gh
+echo "waiting for upload to complete"
+sleep 10
 
-URI=http://rabbitmq:15672
-if [ -n "$PGSSLCERT" ]; then
-   URI=https://rabbitmq:15671
+if [ $((stream_size++)) -eq "$(curl -s -u guest:guest $URI/api/queues/sda/inbox | jq '.messages_ready')" ]; then
+   echo "Upload did not complete successfully"
+   exit 1
```
Comment (aaperis) [inline .github/integration/tests/sda/60_api_admin_test.sh:109]: If it randomly takes more than `sleep 10`, will this fail randomly as well?

File: sda/cmd/mapper/mapper.go [hunk 2]
```diff
@@ -102,14 +103,14 @@ func main() {
 				}
 
 				for _, aID := range mappings.AccessionIDs {
-					log.Debugf("Mapped file to dataset (corr-id: %s, datasetid: %s, accessionid: %s)", delivered.CorrelationId, mappings.DatasetID, aID)
```
Comment (aaperis) [inline sda/cmd/mapper/mapper.go:105]: why remove these logs?

**Review by aaperis — COMMENTED** [review]
Looks good! Some minor comments/questions :-)

File: .github/integration/tests/sda/60_api_admin_test.sh [hunk 3]
```diff
@@ -41,25 +41,25 @@ fi
...
+if [ $((stream_size++)) -eq "$(curl -s -u guest:guest http://rabbitmq:15672/api/queues/sda/inbox | jq '.messages_ready')" ]; then
    echo "waiting for upload to complete"
-   RETRY_TIMES=$((RETRY_TIMES + 1))
-   if [ "$RETRY_TIMES" -eq 30 ]; then
-      echo "::error::Time out while waiting for upload to complete"
-      exit 1
-   fi
-   sleep 2
-done
+   echo "but too bad will eexit"
```
[the remark this answers is not in the document]
Comment (aaperis) [reply .github/integration/tests/sda/60_api_admin_test.sh]: resolved :-)

File: .github/integration/tests/sda/60_api_admin_test.sh (hunk 1 above)
Comment (MalinAhlberg) [reply .github/integration/tests/sda/60_api_admin_test.sh]: Reworked in 3353a1f

File: sda/cmd/mapper/mapper.go (hunk 2 above)
Comment (MalinAhlberg) [reply sda/cmd/mapper/mapper.go:105]: By accident :face_with_peeking_eye: . Good that you noticed. Readded in e263eb3.

**Review by aaperis — APPROVED** [review]
Nice work!

**Pull request merged**